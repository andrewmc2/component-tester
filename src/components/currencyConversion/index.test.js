/* eslint-env jest */

const { shallow } = require('enzyme')
const React = require('react')
const toJSON = require('enzyme-to-json').default
const CurrencyConversion = require('./index').default

describe('<CurrencyConversion />', () => {
  const defaultProps = {
    value: 1e6, // a mil
    to: 'BTC',
    valueOnly: undefined,
    precision: 10,
    commas: undefined
  }

  function buildComponent (props = {}) {
    props = Object.assign({}, defaultProps, props)
    return <CurrencyConversion {...props} />
  }

  it('renders satoshi values to btc', () => {
    const component = shallow(buildComponent())
    expect(toJSON(component)).toMatchSnapshot('currencyConversion')
  })

  it('renders satoshi values to mbtc', () => {
    const component = shallow(buildComponent({ to: 'mBTC', valueOnly: false }))
    expect(toJSON(component)).toMatchSnapshot('currencyConversion')
  })

  it('should allow for only rendering the value', () => {
    const component = shallow(buildComponent({ valueOnly: true }))
    expect(toJSON(component)).toMatchSnapshot('currencyConversion')
  })

  it('should allow for overriding precision', () => {
    const component = shallow(buildComponent({ valueOnly: false, precision: 2 }))
    expect(toJSON(component)).toMatchSnapshot('currencyConversion')
  })

  it('should allow for comma separation', () => {
    const component = shallow(buildComponent({ value: 100e6, to: 'mBTC', valueOnly: false, precision: 0, commas: true }))
    expect(toJSON(component)).toMatchSnapshot('currencyConversion')
  })
})
