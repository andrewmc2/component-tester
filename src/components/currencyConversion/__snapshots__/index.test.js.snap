// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`currencyConversion 1`] = `
<span
  className=""
>
  0.0100000000 BTC
</span>
`;

exports[`currencyConversion 2`] = `
<span
  className=""
>
  10 mBTC
</span>
`;

exports[`currencyConversion 3`] = `
<span
  className=""
>
  0.0100000000
</span>
`;

exports[`currencyConversion 4`] = `
<span
  className=""
>
  0.01 BTC
</span>
`;

exports[`currencyConversion 5`] = `
<span
  className=""
>
  1,000 mBTC
</span>
`;
